### PR TITLE
README / troubleshooting docs improvements

### PR DESCRIPTION
**Pull request by kaoudis**
Hi there! I wanted to clean up the base docs a bit based on my experience getting started with Docker-OSX. 

What's changed
- added some links to various closed issues
- fixed dead links
- unified internal link formatting (#the-section-name allows links to work locally and in forks, not just in this repo)
- moved around sections for flow and resized some headings

If this is a lot, feel free to request changes or modify yourself, otherwise I can split this into multiple PRs by commit :slightly_smiling_face:

### CODE REVIEW

**Comment by kaoudis** [comment]
If you wish to see how this looks before merging, it's up here https://github.com/kaoudis/Docker-OSX/ or this tool is absolutely magical for previewing local Markdown edits just as they'll be on Github: https://github.com/joeyespo/grip

**Review by sickcodes — COMMENTED** [review]
> If you wish to see how this looks before merging, it's up here https://github.com/kaoudis/Docker-OSX/ or this tool is absolutely magical for previewing local Markdown edits just as they'll be on Github: https://github.com/joeyespo/grip

Will check it out, I use a Sublime Text plugin and Press Ctrl B to generate markdown to HTML, then Alt Tab to Chrome and Ctrl R:

Screenshot_2021-04-08_10-02-31

**Comment by sickcodes** [comment]
LGTM!

File: README.md [hunk 1]
```diff
@@ -83,7 +86,7 @@ The images (excluding `:naked`) launch a container with an existing installation
 
 Images built on top of the contents of this repository are also available on **Docker Hub** for convenience: https://hub.docker.com/r/sickcodes/docker-osx
 
-A comprehensive list of the available Docker images and their intended purpose can be found in the [Instructions](README.md#Instructions).
+A comprehensive list of the available Docker images and their intended purpose can be found in the [Instructions](#instructions).
```
[the remark this answers is not in the document]
Comment (kaoudis) [reply README.md:89]: yup, the # references I think have to be lowercase

**Pull request merged**